src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.js

```diff
@@ -258,43 +262,59 @@ angular.module('esn.inbox.libs')
     }
 
     function updateSharedMailboxCache() {
+
       return withJmapClient(function(jmapClient) {
-        return jmapClient.mailbox_get({
+        return jmapClient.mailbox_changes({
           accountId: null,
-          ids: null
-        })
-          .then(function(mailboxes) {
-            return mailboxes.list;
-          })
-          .then(function(mailboxList) {
-            return _addSharedMailboxVisibility(_getSharedMailboxes(mailboxList));
-          })
-          .then(function(sharedMailboxList) {
-            const sharedMailboxCache = _getSharedMailboxes(inboxMailboxesCache);
-            const removedSharedFoldersIds = _getDifferenceById(sharedMailboxCache, sharedMailboxList);
+          sinceState: inboxMailboxesCache.state
+        }).then(function(changes) {
+          if (changes.newState !== inboxMailboxesCache.state) {
+            return jmapClient.mailbox_get({
+              accountId: null,
+              ids: null
+            })
+              .then(function(mailboxes) {
+                _updateMailBoxCacheState(mailboxes.state);
+
+                return mailboxes.list;
+              })
+              .then(function(mailboxList) {
+                return _addSharedMailboxVisibility(_getSharedMailboxes(mailboxList));
+              })
+              .then(function(sharedMailboxList) {
+                return _updateSharedMailboxList(sharedMailboxList);
+              })
+              .then(function() {
+                $rootScope.$broadcast(INBOX_EVENTS.SHARED_FOLDERS_UPDATED);
+
+                return _getSharedMailboxes(inboxMailboxesCache.list);
+              });
+          }
+
+          return _getSharedMailboxes(inboxMailboxesCache.list);
```

Comment:
If those lines are put in a `then` after the changes functions, they would not have to be duplicated.

Reply:
you auto review a part of your code (pair )
what do you mean

Reply:
done